appengine/flexible/Metadata/Metadata.Sample/Program.cs

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+public class Program
+{
+    private WebApplication App {get; set;}
+
+    private Program(string[] args)
+    {
+        var builder = WebApplication.CreateBuilder(args);
+        App = builder.Build();
+        App.MapGet("/", HandleGetAsync);
+    }
+
+    private async Task HandleGetAsync(HttpContext context)
+    {
+        string response;
+        try {
+            // [START gae_flex_metadata]
+            var client = new HttpClient();
+            client.DefaultRequestHeaders.Add("Metadata-Flavor", new[] { "Google" });
+            response = await client.GetStringAsync(
+                "http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip");
+            // [END gae_flex_metadata]
+        }
+        catch (System.Net.Http.HttpRequestException e)
+        {
+            response = "<not set>";
+        }
+        await context.Response.WriteAsync(string.Format(
+            "Public IP address: {0}",
+	    response));
+    }

```

Comment:
I'm wondering if we shouldn't do any of this, in this sample and the others that make an outbound call. Basically, if we should always call `response.EnsureSuccessStatusCode()` so that the sample method throws if there was an error . And tests would fail in that case, whereas they are not failing now because what we are really testing for is that the call `context.Response.WriteAsync` succeeds.

Of course that would mean that all of these samples that need to be actually running in App Engine Flex will fail in local or any other environment. What do you think?

Reply:
I'm open to changing it, but is there a particular reason we'd want to change it? I'm a bit hesitant to iterate on the sample code because it's just supposed to be a sample to start with, anyway.

Reply:
Yes you are right. We should be testing that the sample as a whole works, and that's what we are doing at the moment. It's not for samples to test for actual product features, it's definetely a plus if they do (and we have caught some API issues that way), but product features should be covered by product tests regardless.
Let's leave it as, we can always re-evaluate later.

Reply:
Ok, sounds good. I pinged a serverless-runtimes reviewer this morning, so just waiting on that now.

Reply:
Adding my 2 cents. I agree with the general ideas. The tests in this repo are responsible for guarding the regressions in this repo. Detecting service and server-side regressions is out-of-scope in my opinion. We have unit tests and e2e tests in our CI/CD to prevent regressions and outages. This repo is not the right place to do so, because it's too late in the release process.